Public/Src/Engine/Processes/FileAccessPolicy.cs

```diff
@@ -81,6 +81,19 @@ public enum FileAccessPolicy : ushort
         /// </remarks>
         AllowRealInputTimestamps = 0x200,
 
+        /// <summary>
+        /// Override writes allowed by policy based on file existence checks. 
+        /// </summary>
+        /// <remarks>
+        /// Override writes allowed by policy based on file existence checks. 
+        /// Used mainly in the context of shared opaques, where the whole cone under the opaque root is write-allowed by policy (except known inputs).
+        /// This policy makes sure that writes on undeclared inputs that fall under the write-allowed cone are flagged as DFAs.
+        /// The way to dermine undeclared inputs is based on file existence: if a pip tries to write into a file - allowed by policy - but
+        /// that was not created by the pip (i.e. the file was there before the first write), then it is a write on an undeclared input

```

Comment:
>by the pip [](start = 33, length = 10)

What's the behavior when the file is created by another pip?

Reply:
Same. Under this policy a DFA is generated whenever a pip tries to write on a file that it didn't create itself. That's the shared opaque invariant (all content has rewrite count 1)

---
In reply to: [299179577](https://github.com/microsoft/BuildXL/pull/536#discussion_r299179577) [](ancestors = 299179577)

Reply:
I like the words "a file that it didn't create itself"

---
In reply to: [299187361](https://github.com/microsoft/BuildXL/pull/536#discussion_r299187361) [](ancestors = 299187361,299179577)

Reply:
Is my own dumb attempt at the Russell's paradox :)

---
In reply to: [299196126](https://github.com/microsoft/BuildXL/pull/536#discussion_r299196126) [](ancestors = 299196126,299187361,299179577)